src/main/java/reposense/model/Author.java

```diff
@@ -8,21 +8,91 @@
 public class Author {
     public static final String UNKNOWN_AUTHOR_GIT_ID = "-";
 
+    private static final String MESSAGE_ILLEGAL_GIT_ID = "The provided Git ID, %s, contains illegal characters.";
+    private static final String MESSAGE_ILLEGAL_DISPLAY_NAME =
+            "The provided display name, %s, contains illegal characters.";
+    private static final String MESSAGE_ILLEGAL_AUTHOR_ALIAS =
+            "The provided author alias, %s,  contains illegal characters.";
+    private static final String MESSAGE_UNCOMMON_GLOB_PATTERN =
+            "The provided ignore glob, %s,  uses uncommon pattern.";
+
+    private static final String COMMON_GLOB_REGEX = "^[-a-zA-Z0-9 _/\\\\*!{}\\[\\]!(),:.]*$";
+    private static final String NAME_VALIDATION_REGEX = "^[-a-zA-Z0-9 _/\\\\*]+$";
+
     private String gitId;
 
+    private transient String displayName;
     private transient List<String> authorAliases;
     private transient List<String> ignoreGlobList;
     private transient PathMatcher ignoreGlobMatcher;
 
+
     public Author(String gitId) {
+        if (!isValidName(gitId)) {
+            throw new IllegalArgumentException(String.format(MESSAGE_ILLEGAL_GIT_ID, gitId));
+        }
         this.gitId = gitId;
+        this.displayName = gitId;
+        this.authorAliases = new ArrayList<>();
+        this.ignoreGlobList = new ArrayList<>();
 
-        authorAliases = new ArrayList<>();
-        ignoreGlobList = new ArrayList<>();
+        updateIgnoreGlobMatcher();
+    }
+
+    public Author(StandaloneAuthor sa) {
+        String gitId = sa.getGithubId();
+        String displayName = !sa.getDisplayName().isEmpty() ? sa.getDisplayName() : sa.getGithubId();
+        List<String> authorAliases = sa.getAuthorNames();
+        List<String> ignoreGlobList = sa.getIgnoreGlobList();
+
+        if (!isValidName(gitId)) {
+            throw new IllegalArgumentException(String.format(MESSAGE_ILLEGAL_GIT_ID, gitId));
+        }
+        if (!isValidName(displayName)) {
+            throw new IllegalArgumentException(String.format(MESSAGE_ILLEGAL_DISPLAY_NAME, displayName));
+        }
+        validateAuthorAliases(authorAliases);
+        validateIgnoreGlobs(ignoreGlobList);
+
+        this.gitId = gitId;
+        this.displayName = displayName;
+        this.authorAliases = authorAliases;
+        this.ignoreGlobList = new ArrayList<>(ignoreGlobList);
 
         updateIgnoreGlobMatcher();
     }
 
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    private static boolean isValidName(String name) {
+        return name.matches(NAME_VALIDATION_REGEX);
+    }
+
+    /**
+     * Returns true if all the strings in the {@code ignoreGlobList} only contains commonly used glob patterns.
+     */
+    private static boolean validateIgnoreGlobs(List<String> ignoreGlobList) {
```

Comment:
no need to be a `boolean`?
Instead, should mention that it throws `IllegalArgumentException`?